### DISCUSSION TITLE
Tensorflow support in spacy-transformers

### DISCUSSION THREAD

**Discussion by shaunnarayan** [explosion/spaCy#9082] [New Features & Project Ideas]
The metal pluggable device means we can run accelerated models on macs now, and the transformer model is quite slow on my M1 mac via PyTorch CPU.

From my understanding spacy-transformers uses thinc which has abstractions for both tensorflow and pytorch, but from what I can tell so far the spacy-transformers layer only supports PyTorch - is that correct?

**Comment by svlandeg** [comment]
Hi,

Your understanding is correct. The [`TransformerModel`](https://github.com/explosion/spacy-transformers/blob/master/spacy_transformers/layers/transformer_model.py) currently implemented by `spacy-transformers` uses Thinc's `PyTorchWrapper` to wrap the HuggingFace transformer in.

Thinc does have a similar [wrapper](https://thinc.ai/docs/usage-frameworks) for Tensorflow, but it's not supported (yet?) in `spacy-transformers`.

**Reply by shaunnarayan** [reply]
Hi,

Thanks for the insight - I'll look into how feasible it is for me to replace that wrapper.

Appreciate the whole team and your work on this great library.

**Reply by svlandeg** [reply]
Thanks for the nice words!

If you do get this to work, we'd definitely be interested, so feel free to keep us updated with what you find ;-)
